### PR TITLE
chore: remove long jobs

### PR DESCRIPTION
**Pull request by OrKoN**
Let's try to keep any job within 15 min. Also, let's keep installation tests only for the latest node LTS version.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -266,8 +273,6 @@ jobs:
           - macos-latest
           - windows-latest
         node:
-          - 14
-          - 16
           - 18
```
Comment (Lightning00Blade) [inline .github/workflows/ci.yml:286]: Some leftover code for a fix in window issue further down.

Comment (OrKoN) [reply .github/workflows/ci.yml:286]: what do you mean?

Comment (Lightning00Blade) [reply .github/workflows/ci.yml:286]: https://github.com/puppeteer/puppeteer/blob/main/.github/workflows/ci.yml#L297

Comment (OrKoN) [reply .github/workflows/ci.yml:286]: That’s okay. Maybe we still bring this change back if we get regressions. That workaround can be removed once we remove node 14.

**Review by Lightning00Blade — APPROVED** [review]

**Pull request merged**